lib/membrane_live/tokens.ex

```diff
@@ -34,23 +32,36 @@ defmodule MembraneLive.Tokens do
     |> then(&%{"pem" => &1})
   end
 
+  @spec auth_encode(any) :: {:error, atom | keyword} | {:ok, binary, %{optional(binary) => any}}
   def auth_encode(user_id) do
-    signer = Joken.Signer.create("HS256", @auth_secret)
+    signer = create_auth_signer()
     AuthToken.generate_and_sign(%{"user_id" => user_id}, signer)
   end
 
+  @spec auth_decode(binary) :: {:error, atom | keyword} | {:ok, %{optional(binary) => any}}
   def auth_decode(jwt) do
-    signer = Joken.Signer.create("HS256", @auth_secret)
+    signer = create_auth_signer()
     AuthToken.verify_and_validate(jwt, signer)
   end
 
+  @spec refresh_encode(any) ::
+          {:error, atom | keyword} | {:ok, binary, %{optional(binary) => any}}
   def refresh_encode(user_id) do
-    signer = Joken.Signer.create("HS256", @refresh_secret)
+    signer = create_refresh_signer()
     RefreshToken.generate_and_sign(%{"user_id" => user_id}, signer)
   end
 
+  @spec refresh_decode(binary) :: {:error, atom | keyword} | {:ok, %{optional(binary) => any}}
   def refresh_decode(jwt) do
-    signer = Joken.Signer.create("HS256", @refresh_secret)
+    signer = create_refresh_signer()
     RefreshToken.verify_and_validate(jwt, signer)
   end
+
+  defp create_auth_signer(),
+    do: create_signer(Application.fetch_env!(:membrane_live, :token_auth_secret))
+
+  defp create_refresh_signer(),
+    do: create_signer(Application.fetch_env!(:membrane_live, :token_refresh_secret))
+
+  defp create_signer(secret), do: Joken.Signer.create("HS256", secret)
```

Comment:
Refactor it with the use of pipelines.
https://github.com/christopheradams/elixir_style_guide#pipe-operator